src/main/java/com/faforever/client/tournament/game/IsReadyForGameController.java

```diff
@@ -0,0 +1,101 @@
+package com.faforever.client.tournament.game;
+
+import com.faforever.client.fx.Controller;
+import com.faforever.client.i18n.I18n;
+import com.faforever.client.ui.progress.RingProgressIndicator;
+import javafx.animation.KeyFrame;
+import javafx.animation.Timeline;
+import javafx.event.ActionEvent;
+import javafx.scene.Parent;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.layout.VBox;
+import javafx.util.StringConverter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.VisibleForTesting;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class IsReadyForGameController implements Controller<Parent> {
+  private final I18n i18n;
+  public VBox root;
+  public Label description;
+  public RingProgressIndicator progressIndicator;
+  public Button isReadyButton;
+  private int timeLeft;
+  @VisibleForTesting
+  Timeline queuePopTimeUpdater;
+  @Setter
+  private Runnable readyCallback;
+  @Setter
+  private Runnable dismissCallBack;
+
+
+  @Override
+  public void initialize() {
+    progressIndicator.setProgressLableStringConverter(new StringConverter<>() {
+      @Override
+      public String toString(Integer object) {
+        return i18n.number(timeLeft);
```

Comment:
Yes I know it is strange I know XD

The idea is the following:
The Controller calculates the timeLeft till timeout. It then calculates the progress in percent and sets the progress. The passed number is the progress in percent. Why calculate the time left in seconds back from the percentage if we already have it. Also might cause rounding errors :thinking:

Reply:
I guess so but then it seems like it would be best for the widget to have two properties and components. The label and the progress bar. Seems a little odd they are combined to be the same